network/src/p2p/handler.rs

```diff
@@ -1129,15 +1141,25 @@ impl IoHandler<Message> for Handler {
                             is_inbound: false,
                         })?;
                         ctrace!(NETWORK, "Outgoing connect({}) established", stream);
+                        None
                     } else {
                         self.routing_table.remove(con.peer_addr());
                         ctrace!(NETWORK, "Outgoing connect({}) removed", stream);
+                        None
                     }
                 } else {
                     cdebug!(NETWORK, "Invalid outgoing token({}) on deregister", stream);
+                    None
                 }
             }
             _ => unreachable!("Unexpected stream on deregister: {}", stream),
+        };
+
+        if let Some(_address) = peer_socket_adress {
```

Comment:
I don't think using an Option for this is a good indication of intent.
How about making this code a function instead of using `peer_socket_adress`?

Reply:
Can you provide more details? Do you think it is better to put a function in the first and second match?

Reply:
```rust
            let batch = DBTransaction::new();
            let remove_target = peer_socket_adress;
            let s = rlp::encode(&remove_target);
            self.peer_db.delete(batch, &s);
```
I meant extracting the code above into a method and calling the method at line 1065 and 1084.

Reply:
Done